util/deephash/pointer.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package deephash
+
+import (
+	"net/netip"
+	"reflect"
+	"time"
+	"unsafe"
+)
+
+// unsafePointer is an untyped pointer.
+// It is the caller's responsibility to call operations on the correct type.
+type unsafePointer struct{ p unsafe.Pointer }
```

Comment:
can you include a list of what types of untyped pointers are used in this package? there aren't more than a half dozen I think?

for memhashable types, to the value itself.
for slices, ... to the slice header or to the first value?
for arrays, to the first value,
etc